BitwardenShared/Core/Platform/Services/AuthenticatorSyncService.swift

```diff
@@ -0,0 +1,237 @@
+
+    /// The keychain repository for managing the key shared between the PM and Authenticator apps.
+    private let sharedKeychainRepository: SharedKeychainRepository
+
+    /// The service used by the application to manage account state.
+    private let stateService: StateService
+
+    /// a Task that subscribes to the sync setting publisher for accounts. This allows us to take action once
+    /// a user opts-in to Authenticator sync.
+    private var syncSettingSubscriberTask: Task<Void, Error>?
+
+    /// The service used by the application to manage vault access.
+    private let vaultTimeoutService: VaultTimeoutService
+
+    // MARK: Initialization
+
+    /// Initialize a `DefaultAuthenticatorSyncService`.
+    ///
+    /// - Parameters:
+    ///   - authBridgeItemService: The service for managing sharing items to/from the Authenticator app.
+    ///   - cipherService: The service used to manage syncing and updates to the user's ciphers.
+    ///   - clientService: The service that handles common client functionality such as encryption and decryption.
+    ///   - configService: The service to get server-specified configuration.
+    ///   - errorReporter: The service used by the application to report non-fatal errors.\ organizations.
+    ///   - notificationCenterService: Notification Center Service to subscribe to foreground/background events.
+    ///   - sharedKeychainRepository: The keychain repository for managing the key shared
+    ///     between the PM and Authenticator apps.
+    ///   - stateService: The service used by the application to manage account state.
+    ///   - vaultTimeoutService: The service used by the application to manage vault access.
+    ///
+    init(
+        authBridgeItemService: AuthenticatorBridgeItemService,
+        cipherService: CipherService,
+        clientService: ClientService,
+        configService: ConfigService,
+        errorReporter: ErrorReporter,
+        notificationCenterService: NotificationCenterService,
+        sharedKeychainRepository: SharedKeychainRepository,
+        stateService: StateService,
+        vaultTimeoutService: VaultTimeoutService
+    ) {
+        self.authBridgeItemService = authBridgeItemService
+        self.cipherService = cipherService
+        self.clientService = clientService
+        self.configService = configService
+        self.errorReporter = errorReporter
+        self.sharedKeychainRepository = sharedKeychainRepository
+        self.notificationCenterService = notificationCenterService
+        self.stateService = stateService
+        self.vaultTimeoutService = vaultTimeoutService
+        super.init()
+    }
+
+    // MARK: Public Methods
+
+    public func start() {
+        Task {
+            if await configService.getFeatureFlag(FeatureFlag.enableAuthenticatorSync,
+                                                  defaultValue: false) {
+                subscribeToAppState()
+            }
+        }
+    }
```

Comment:
🤔 Should a check be added here to avoid `start` being called multiple times thus subscribing multiple times to  app state? Additionally, should this service be an `actor` to prevent multiple thread trying to call `start` concurrently?

Reply:
The original goal of adding a start() method was to get async out the `ServiceContainer` (because we have to `await` the configService call). So `ServiceContainer` could just call `start()` and move on and then the service would start up asynchronously in the background. Making it an actor means that it implicitly needs to become async and thus the `ServiceContainer` needs to wrap this in a `Task` to let it run without blocking.

However, now that you've mentioned it, I think that's probably the safest way to go. We're relying on the compiler instead of just our own code conventions of not calling it multiple times. And without making it an `actor`, I think it's hard to guarantee that our check for !started is doing what we think it's doing.

I've pushed up a commit that moves this over to an `actor` and adds a check to see if it's started 👍